### PR TITLE
Correct handling of paths with spaces and ampersands when building on windows

### LINKED ISSUE #178: Cannot use build.cmd from a path that contains a space
I could not using "build.cmd clean" from a path that contained a space.

```
'Studio' is not recognized as an internal or external command,
operable program or batch file.
A subdirectory or file C:\Users\Jonathan\Documents\Visual already exists.
Error occurred while processing: C:\Users\Jonathan\Documents\Visual.
'Studio' is not recognized as an internal or external command,
operable program or batch file.
A subdirectory or file C:\Users\Jonathan\Documents\Visual already exists.
Error occurred while processing: C:\Users\Jonathan\Documents\Visual.
A subdirectory or file Studio already exists.
Error occurred while processing: Studio.
'Studio' is not recognized as an internal or external command,
operable program or batch file.
A subdirectory or file C:\Users\Jonathan\Documents\Visual already exists.
Error occurred while processing: C:\Users\Jonathan\Documents\Visual.
A subdirectory or file Studio already exists.
Error occurred while processing: Studio.
'Studio' is not recognized as an internal or external command,
operable program or batch file.
Checking pre-requisites...
```
(I was using C:\Users\Jonathan\Documents\Visual Studio 2013\Projects\coreclr.)

When moved to a path that does not contain a space, the project builds.

### PR DESCRIPTION
**Pull request by poizan42**
Fixes #178

### CODE REVIEW

**Comment by poizan42** [comment]
There is still a pre-build event in cee_wks that fails

File: build.cmd [hunk 1]
```diff
@@ -6,13 +6,15 @@ set __BuildArch=x64
...
 
 :: Set the various build properties here so that CMake and MSBuild can pick them up
-set __ProjectDir=%~dp0
-set __ProjectFilesDir=%~dp0
-set __SourceDir=%__ProjectDir%\src
-set __PackagesDir=%__SourceDir%\.nuget
-set __RootBinDir=%__ProjectDir%\binaries
-set __LogsDir=%__RootBinDir%\Logs
-set __CMakeSlnDir=%__RootBinDir%\CMake
+set "__ProjectDir=%~dp0"
```
Comment (AlexGhiondea) [inline build.cmd]: Should this be set __ProjectDir="%~dp0" instead?

Comment (poizan42) [inline build.cmd]: No it's correct. Otherwise we get the quotes inside the environment variable which wreaks havoc if we try to quote it again later.

Comment (AlexGhiondea) [inline build.cmd]: Makes sense.

**Comment by AlexGhiondea** [comment]
This LGTM but I would like to understand what was that pre-build event that was failing before merging this.

**Comment by poizan42** [comment]
mscorlib is still failing, I'm working on an update

**Comment by poizan42** [comment]
Everything seems to work now with an evil path, except for generating the development NuGet package. The problem is that it seems to expect mscorlib.pdb to be in the same directory as mscorlib.dll, but it has been moved to the PDB subdir. But I'm a bit confused about this as src\.nuget\Microsoft.DotNet.CoreCLR.Development.nuspec does specify this, but we have the task MovePDB in build.proj which moves the mscorlib.pdb file to the PDB directory. Which is the correct behaviour here?

File: src/mscorlib/GenerateSplitStringResources.targets [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <ResGenCommand Condition="'$(ResGenCommand)'==''">$(FrameworkSDKDir)bin\NETFX 4.5.1 Tools\ResGen.exe</ResGenCommand>
+    <ResGenCommand Condition="'$(ResGenCommand)'==''">$(SDK40ToolsPath)ResGen.exe</ResGenCommand>
```
Comment (AlexGhiondea) [inline src/mscorlib/GenerateSplitStringResources.targets]: This does not seem related.

I will have to check, but I *think* we need this to be using the FrameworkSDKDir to allow us to build in the internal mirror.

Comment (poizan42) [inline src/mscorlib/GenerateSplitStringResources.targets]: Yeah your're right. I needed the chance to build mscorlib alone, but that's another issue. I'm taking it out.

**Comment by poizan42** [comment]
This builds successfully from a completely clean state from a path both containing and ampersand and a space. The mscorlib.pdb issue seems to be an unrelated issue with rebuilding the nuget package,

**Comment by AlexGhiondea** [comment]
LGTM.

**Comment by AlexGhiondea** [comment]
@mmitche do you want to take a look?

**Comment by mmitche** [comment]
LGTM

**Pull request merged**